### ISSUE TITLE
Revoking sessions for Desktop App gives "invalid session" errors in server logs

### ISSUE THREAD

**Issue by jasonblais** [mattermost/desktop#368]
I confirm (by marking "x" in the [ ] below): 

- [X] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [X] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [X] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md). 

---

**Summary**

Revoking sessions for Desktop App gives "invalid session" errors in server logs

**Steps to reproduce**

Environment:
- Windows 10 / Mattermost Server v3.5.0 / Mattermost Desktop App v3.4.1

Repro steps:
1) In a Mattermost server, go to **Account Settings > Security > View and Logout of Active Sessions**
2) Find the session corresponding to the desktop app
3) Hit "Logout"
4) With a System Admin account, go to **System Console > Logs** and find the most recent errors

**Expected behavior**

No errors pertaining to revoking a desktop app session

**Observed behavior**

The following errors show up in Logs for each revoked desktop app session:

```
[2016/11/22 22:17:58 UTC] [EROR] Invalid session token=cyy5izi1jbrnzcdrcjxoqe1rar, err=sessionIdOrToken=cyy5izi1jbrnzcdrcjxoqe1rar
[2016/11/22 22:17:58 UTC] [EROR] /api/v3/users/status/set_active_channel:ServeHTTP code=401 rid=53pbqf9xg3fntqrguq9kgf3fbc uid= ip=XXXXXXXXXX Invalid or expired session, please login again. [details: token=cyy5izi1jbrnzcdrcjxoqe1rar]
```

**Possible fixes**

?

**Comment by jnugh** [comment]
Just tried the following:
I just tried this with browsers (e.g. login in two different browsers or in an incognito tab or something) and Log this session out from the other Browser.

The messages in the system log are slightly different from yours, but this might just be related to timing (your client tried to call a different API than mine).

The only fix I can see is to send a message to the client that has just been logged off (if it was active) so that it can react to that by suspending eventual API requests...

**Comment by jasonblais** [comment]
Thanks @jnugh - would the proposed fix remove the errors from the System logs?

Would a message be displayed on the JavaScript console?

Any side effects that you might foresee to the user?

**Comment by jnugh** [comment]
This is something which needs to be fixed in platform - the logout has to notify the browser/app which runs the session. Nothing we can do (and not limited to the desktop app)

**Comment by jasonblais** [comment]
Ah, got it. I'll let the platform team know and create a ticket for them

**Comment by jasonblais** [comment]
Ticket is here: https://mattermost.atlassian.net/browse/PLT-4940

We might still want to consider fixing it ourselves and potentially submitting a PR for it. I'm not sure when the platform team would get to this bug..

If there's any information that would be useful to fix it, let me know.

**Comment by jasonblais** [comment]
Closing - this will be tracked on the platform side and a fix is scheduled for Mattermost Server 3.7

Thanks @jnugh for your help!
